src/lyra/abstract_domains/numerical/interval_domain.py

```diff
@@ -42,7 +42,7 @@ class IntervalLattice(BottomMixin, ArithmeticMixin, BooleanMixin, SequenceMixin)
 
     def __init__(self, lower=-inf, upper=inf):
         super().__init__()
-        if lower <= upper:      # the interval is not empty
+        if lower <= upper and lower != inf and upper != -inf:      # the interval is not empty
```

Comment:
Yeah, I am not happy about this. Intervals like `[-inf, -inf]` and `[inf, inf]` are not empty.

Reply:
But are they valid intervals? You have to deal with these cases somehow, don't you?

Reply:
Mh, well, looking again at the formal definitions of the domain `lower` is generally not allowed to have value 'inf' and upper is generally not allowed to have value '-inf' so I guess I can survive with this change ;)

Reply:
You could also raise an exception if you want.